Automattic/jetpack
# Blogroll: cleanup appending state mgmt

This a pure refactor to be a first step to implement, https://github.com/Automattic/wp-calypso/issues/82003.

## Proposed changes:
1. This simplifies and modulizes the site fetching logic.
2. Makes debouncing only against network requests.
3. Prefers cached responses over fresh ones.

After this PR, it will be much simpler to support custom links.

Frankly, I started with the ambition to support custom links, but decided to send this easy PR first. That change will need a of changes.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Add the blogroll.
4. Try your best to break the appender.
5. If you search for a non-domain, a request should not be issued.
6. If you search for a domain, you should see "loading...", then the result.
7. Search should be debounced by 500ms.
8. If you research an already fetched item, the request should be issued, but should say "disk cache" in DevTools.
9. If you simulate slow 3G, search quickly two domains, the previous request should be canceled.
